integration/common/tests/dbt/structured_logs/test_structured_logs.py

```diff
@@ -0,0 +1,210 @@
+# Copyright 2018-2024 contributors to the OpenLineage project
+# SPDX-License-Identifier: Apache-2.0
+import json
+from enum import Enum
+from typing import Dict
+
+import attr
+import pytest
+from openlineage.common.provider.dbt.processor import Adapter
+from openlineage.common.provider.dbt.structured_logs import DbtStructuredLogsProcessor
+from openlineage.common.test import match
+
+###########
+# helpers
+###########
+
+
+def ol_event_to_dict(event) -> Dict:
+    return attr.asdict(event, value_serializer=serialize)
+
+
+def serialize(inst, field, value):
+    if isinstance(value, Enum):
+        return value.value
+    return value
+
+
+##################
+# fixtures
+##################
+
+
+@pytest.fixture(autouse=True)
+def patch_get_dbt_profiles_dir(monkeypatch):
+    monkeypatch.setattr(
+        "openlineage.common.provider.dbt.structured_logs.get_dbt_profiles_dir",
+        lambda *args, **kwargs: "./tests/dbt/structured_logs",
+    )
+
+
+##################
+# test functions
+##################
+
+
+@pytest.mark.parametrize(
+    "target, command_line, logs_path, expected_ol_events_path, manifest_path",
+    [
+        # successful postgres run
+        (
+            "postgres",
+            ["dbt", "run", "..."],
+            "./tests/dbt/structured_logs/postgres/run/logs/successful_run_logs.jsonl",
+            "./tests/dbt/structured_logs/postgres/run/results/successful_run_ol_events.json",
```

Comment:
have a look at the results json file to see the structure of the OL events generated based on the logs and manifest.